### PR TITLE
Bmc ldap

### PR DESCRIPTION
**Pull request by myMarck**
Check if you agree with the changes

### CODE REVIEW

File: lib/puppet/provider/bmc_ldap/racadm7.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  #candiate to be moved to a shared lib
+  def racadm_call cmd_args
+    cmd = ['/opt/dell/srvadmin/bin/idracadm']
+    cmd.push('-u').push(resource[:username]) if resource[:username]
+    cmd.push('-p').push(resource[:password]) if resource[:password]
+    if resource[:bmc_server_host]
+      cmd.push('-r').push(resource[:bmc_server_host])
+    else
+      ipmitool_out = ipmitool('lan', 'print')
+      lanPrint = Ipmi::Ipmitool.parseLan(ipmitool_out)
```
Comment (enemarke) [inline lib/puppet/provider/bmc_ldap/racadm7.rb]: could just be 
lanPrint = Ipmi::Ipmitool.parseLan(ipmitool('lan', 'print'))

File: lib/puppet/provider/bmc_ldap/racadm7.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      cmd.push('-r').push(lanPrint['IP Address'])
+    end
+
+    command = cmd + cmd_args
```
Comment (enemarke) [inline lib/puppet/provider/bmc_ldap/racadm7.rb]: why not 
cmd.push(cmd_args)
?

File: lib/puppet/provider/bmc_ssl/racadm7.rb [hunk 3]
```diff
@@ -63,8 +62,8 @@ def racadm_call cmd_args
     cmd = ['/opt/dell/srvadmin/bin/idracadm']
     cmd.push('-u').push(resource[:username]) if resource[:username]
     cmd.push('-p').push(resource[:password]) if resource[:password]
-    if resource[:remote_rac_host]
-      cmd.push('-r').push(resource[:remote_rac_host])
+    if resource[:bmc_server_host]
+      cmd.push('-r').push(resource[:bmc_server_host])
     else
       ipmitool_out = ipmitool('lan', 'print')
```
Comment (enemarke) [inline lib/puppet/provider/bmc_ssl/racadm7.rb]: maybe just 
lanPrint = Ipmi::Ipmitool.parseLan(ipmitool_out('lan', 'print'))

File: lib/puppet/type/bmc_network.rb [hunk 4]
```diff
@@ -20,36 +20,46 @@
   newproperty(:ipaddr) do
     desc 'Ip Address'
     validate do |value|
-      raise ArgumentError, "%s is not a valid ip address" % value unless (value =~ Resolv::IPv4::Regex || value =~ Resolv::IPv6::Regex)
+      unless (value =~ Resolv::IPv4::Regex || value =~ Resolv::IPv6::Regex)
+        raise Puppet::Error, "%s is not a valid ip address" % value
+      end
```
Comment (enemarke) [inline lib/puppet/type/bmc_network.rb]: Create a define and call it. You are using the same unless statement to raise and error four times.

File: spec/unit/puppet_x/racadm_spec.rb [hunk 5]
```diff
@@ -4,14 +4,20 @@
...
 
-  context 'racadm7 get IDRAC.IPv4' do
-    subject { Racadm::Racadm.parseiDRAC_IPv4 getIDRAC }
+  let(:getIDRAC_LDAP) { IO.read(
+      File.join(
+          File.dirname(__FILE__),
+          '..', '..', 'fixtures', 'bmc', 'racadm7_getIDRAC.LDAP.txt'))
+  }
+
+  describe 'racadm7 get IDRAC.IPv4' do
```
Comment (enemarke) [inline spec/unit/puppet_x/racadm_spec.rb]: use context here. context is an alias for describe, but the standard says to use it like:
describe ... do
  context ... do
  end
end

File: lib/puppet/provider/bmc_ldap/racadm7.rb [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      @property_hash[:group_attribue] = iDRAC_LDAP['GroupAttribute']
+      @property_hash[:search_filer] = iDRAC_LDAP['SearchFilter']
+      @property_hash[:certificate_validate] = (iDRAC_LDAP['CertValidationEnable'].eql? 'Enabled').to_s
+      @property_hash[:group_attribute_is_dn] = (iDRAC_LDAP['GroupAttributeIsDN'].eql? 'Enabled').to_s
+      @property_hash[:ensure] = (iDRAC_LDAP['Enable'].eql? 'Enabled') ? :present : :absent
+    end
+    @property_hash[:ensure] == :present
+  end
+
+  def server= value
```
Comment (enemarke) [inline lib/puppet/provider/bmc_ldap/racadm7.rb]: when defining setters should it be:
```
def server= value
```
or
```
def server=value
```
?
I prefer the last one.

Comment (myMarck) [inline lib/puppet/provider/bmc_ldap/racadm7.rb]: Ok - I will change all to follow the "der server=value" pattern

**Pull request merged**